### DISCUSSION TITLE
Include `fb:app_id` in `generateMetadata`

### DISCUSSION THREAD

**Discussion by bryanjhickey** [vercel/next.js#47111] [Ideas]
### Goals

1. Include `fb:app_id` in `generateMetadata`, possibly under `openGraph` or (better?) `verification`?
2. Include more flexibility in `custom` options because we can't pass `fb:app_id` in the `name`: `content` value pair as facebook expects `name` for `app_id` to be passed as `property`

### Non-Goals

1.
2.
3.

### Background

I'm not aware of any alternatives to the approaches IO have discussed above. 

### Proposal

Update type definitions for and `generateMetadata` tooling to include this option.

**Comment by ciruz** [comment]
You can also add custom meta tags, like so:

```javascript
export const metadata = {
  title: "Home",
  description: "Welcome to Next.js",
  viewport:
    "width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0",
  other: {
    ["fb:app_id"]: "your_app_id",
  },
};
```

Result:
![image](https://user-images.githubusercontent.com/987947/225014018-421a6911-8130-4b75-9169-f623a2750275.png)

Ah I see, in this case it would be useful as `property` not as `name`, but I leave my example for others.

**Reply by benkingcode** [reply]
It's invalid with `name`

**Reply by piercing-wind** [reply]
hello , can you help me on solving the problem related to the google search engine, my website name is appearing as domain name not as AI Flavoured on google search, 

i have explained it on stackover flow in deatil if need further reference.
https://stackoverflow.com/questions/78803518/how-to-display-website-name-on-google-search-engine-with-nginx-next-js-14-appl 

![Screenshot from 2024-07-28 14-43-01](https://github.com/user-attachments/assets/e0ee7ed8-c1bd-40ae-b614-bb8ecb83f412)
here is the reference image

**Reply by vigilproctorplus** [reply]
but how to get the output like this property parameter instated of name parameter. <meta property="fb:app_id" content="1234">

**Reply by Protik111** [reply]
Could you please share how can I add property instead of name?

**Reply by Protik111** [reply]
> It's invalid with `name`

Did you make it 'property' replacing the 'name'?

**Comment by PawelPyc154** [comment]
@bryanjhickey try
```
const RootLayout = async ({ children }: RootLayoutProps) => (
  <html>
    <head>
      <meta property="fb:app_id" content="" />
    </head>
    <body>{children}</body>
  </html>
)

export default RootLayout
```

**Reply by jehoshua-shey** [reply]
This worked for me

nextjs 13.5.4

**Reply by g-popovic** [reply]
This might work, however keep in mind that [Next.js Docs](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#metadata) explicitly state that the `<head>` element should not be manually manipluated. There is [another idea](https://github.com/vercel/next.js/discussions/57527) which is still open that is also related to this particular issue though
